go/builder.go

```diff
@@ -53,6 +53,10 @@ func (b *Builder) Reset() {
 		b.vtable = b.vtable[:0]
 	}
 
+	if b.sharedStrings != nil {
+		b.sharedStrings = map[string]UOffsetT{}
+	}
+
```

Comment:
I am not sure who review Go, but I googled how to reset a map in Go lang and found the following
```go
// Method - I , say book is name of map
for k := range book {
    delete(book, k)
}

// Method - II
book = make(map[string]int)

// Method - III
book = map[string]int{}
```

Reply:
@rw wrote the Go port, but yeah the last 2 options seem nicest.

Reply:
AFAICT Option (1) keeps the memory, possibly reducing heap allocations if the map is re-used.

Reply:
@rw in CSharp apparently we would want to null the memory so should we have different implementation for different languages?